app/components/progressive_image/progressive_image.test.js

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {shallow} from 'enzyme';
+
+import Preferences from 'mattermost-redux/constants/preferences';
+
+import ProgressiveImage from './progressive_image';
+
+describe('ProgressiveImage', () => {
+    const baseProps = {
+        isBackgroundImage: false,
+        defaultSource: 0,
+        filename: 'defaultImage',
+        imageUri: 'https://images.com/image.png',
+        onError: jest.fn(),
+        resizeMethod: 'auto',
+        resizeMode: 'contain',
+        theme: Preferences.THEMES.default,
+        thumbnailUri: 'https://images.com/image.png',
+        tintDefaultSource: false,
+    };
+
+    test('should match snapshot when just a image loads', () => {
+        const wrapper = shallow(
+            <ProgressiveImage {...baseProps}/>
+        );
+        expect(wrapper.getElement()).toMatchSnapshot();
+    });
+
+    test('should match snapshot when no thumbnail', () => {
+        const props = {
+            ...baseProps,
+            thumbnailUri: null,
+        };
+
+        const wrapper = shallow(
+            <ProgressiveImage {...props}/>
+        );
+        expect(wrapper.getElement()).toMatchSnapshot();
+    });
```

Comment:
I feel like these do not make any difference when tested with snapshot only.  Would be nice if something could be checked if a function is called  (or called with what), matches state, etc.

Non-blocking but feel free to add if it makes sense.

Reply:
Will be creating a subtask for this to resolve the unit tests